src/resources/ethernet_esp32.cc

```diff
@@ -15,7 +15,7 @@
 
 #include "../top.h"
 
-#if defined(TOIT_FREERTOS) && defined(CONFIG_TOIT_ENABLE_ETHERNET)
+#if defined(TOIT_FREERTOS) && defined(CONFIG_IDF_TARGET_ESP32) && defined(CONFIG_TOIT_ENABLE_ETHERNET)
```

Comment:
This doesn't work on the S3 and instead of failing from within init_esp32, it now is just a bunch of unimplemented primitives?

Reply:
Maybe we can add dependence(IDF_TARGET_ESP32) for `TOIT_ENABLE_ETHERNET` in Kconfig.

Reply:
reverted.